### PR TITLE
Assertions automateDataPreparation

### PR DESCRIPTION
**Pull request by grewered**

### CODE REVIEW

File: R/automateDataPreparation.R [hunk 1]
```diff
@@ -7,6 +7,22 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
 						addLeadingZeros=FALSE, truncateSpaceChar = TRUE, newID = NULL, oldIDs = NULL,
             missing.rule = list(mvi = 0, mnr = 0, mci = NA, mbd = NA, mir = 0, mbi = 0), verbose=FALSE) {
 
+  checkmate::assert_list(datList, types = "data.frame", null.ok = TRUE)
+  checkmate::assert_list(inputList, types = "data.frame")
+  checkmate::assert_character(path, len = 1, null.ok = TRUE)
```
Comment (grewered) [inline R/automateDataPreparation.R:12]: `path` has `NULL` as default, but in man page line 24 it says 'current working directory' is the default. I you want, I could correct that, but you'd have to tell me which one is correct.

File: R/automateDataPreparation.R [hunk 2]
```diff
@@ -7,6 +7,22 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
...
+  lapply(c(readSpss, checkData, mergeData, recodeData, recodeMnr,
+           aggregateData, scoreData, writeSpss, collapseMissings, suppressErr,
+           rename, recodedData, addLeadingZeros, truncateSpaceChar, verbose), checkmate::assert_logical, len = 1)
+  checkmate::assert_numeric(breaks, null.ok = TRUE)
+  checkmate::assert_numeric(nMbi, len = 1, lower = 1)
+  checkmate::assert_scalar(rotation.id, null.ok = TRUE)
+  checkmate::assert_character(recodeErr, len = 1)
+
+  checkmate::assert_character(newID, len = 1)
+  checkmate::assert_character(oldIDs, null.ok = TRUE)
```
Comment (grewered) [inline R/automateDataPreparation.R:23]: IDs as character, or as vector? man page says character. but in the other functions it was often character or numeric, right?

**Comment by grewered** [comment]
tests in testthat file fail, because:

grafik
Fehler in automateDataPreparation(inputList = NULL, datList = NULL, path = "none",  : Assertion on 'inputList' failed: Must be of type 'list', not 'NULL'.
- shall I allow `NULL` Input for `ìnputList`? or write a different error message to compare to?

**Comment by grewered** [comment]
> > tests in testthat file fail, because:
> > grafik Fehler in automateDataPreparation(inputList = NULL, datList = NULL, path = "none", : Assertion on 'inputList' failed: Must be of type 'list', not 'NULL'.
> > ```
> > * shall I allow `NULL` Input for `ìnputList`? or write a different error message to compare to?
> > ```
> 
> if inputList is NULL, datList is not allowed to be NULL, too; in this case, datList must be a data.frame or a list of data.frames

which means I shall do what? change the assertions in the function? or change the testthat test?

**Comment by sachseka** [comment]
> which means I shall do what? change the assertions in the function? or change the testthat test?

in this case please leave the test as is and change the assertions, that is, allow NULL for inputList. optional: check if there is already a check in the function that datList must be non-empty when inputList is NULL (I believe there is one but I will check this, too, when reviewing later)

**Comment by grewered** [comment]
> > which means I shall do what? change the assertions in the function? or change the testthat test?
> 
> in this case please leave the test as is and change the assertions, that is, allow NULL for inputList. optional: check if there is already a check in the function that datList must be non-empty when inputList is NULL (I believe there is one but I will check this, too, when reviewing later)

there seems to be a check for that starting in line 55

File: R/automateDataPreparation.R [hunk 3]
```diff
@@ -7,6 +7,22 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
 						addLeadingZeros=FALSE, truncateSpaceChar = TRUE, newID = NULL, oldIDs = NULL,
             missing.rule = list(mvi = 0, mnr = 0, mci = NA, mbd = NA, mir = 0, mbi = 0), verbose=FALSE) {
 
+  checkmate::assert_list(datList, types = "data.frame", null.ok = TRUE)
```
Comment (sachseka) [inline R/automateDataPreparation.R]: this check is too restrictive. It should be allowed that datList is just a single data.frame. When I do this I get an error:
Fehler: Assertion on 'singeDat' failed: Must be of type 'list' (or 'NULL'), not 'data.frame'. I added a test for this scenario that must pass.

File: R/automateDataPreparation.R [hunk 4]
```diff
@@ -7,6 +7,22 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
...
+  checkmate::assert_list(inputList, types = "data.frame", null.ok = TRUE)
+  checkmate::assert_character(path, len = 1, null.ok = TRUE)
+  lapply(c(filedat, filesps), checkmate::assert_character, len = 1)
+  lapply(c(readSpss, checkData, mergeData, recodeData, recodeMnr,
+           aggregateData, scoreData, writeSpss, collapseMissings, suppressErr,
+           rename, recodedData, addLeadingZeros, truncateSpaceChar, verbose), checkmate::assert_logical, len = 1)
+  checkmate::assert_numeric(breaks, null.ok = TRUE)
+  checkmate::assert_numeric(nMbi, len = 1, lower = 1)
+  checkmate::assert_scalar(rotation.id, null.ok = TRUE)
```
Comment (sachseka) [inline R/automateDataPreparation.R:19]: rotation.id must be 
Character vector of length 1 (documentation, passed on to function mnrCoding indicating the name of the rotation indicator (e.g. “booklet”) in the dataset.)

File: R/automateDataPreparation.R (hunk 2 above)
Comment (sachseka) [reply R/automateDataPreparation.R:23]: yes but like rotation.id here we require character

**Review by sachseka — CHANGES_REQUESTED** [review]
sorry, still not ready, but the majority already looks very good!

File: R/automateDataPreparation.R [hunk 5]
```diff
@@ -7,6 +7,23 @@ automateDataPreparation <- function(datList = NULL, inputList, path = NULL,
...
+  lapply(c(readSpss, checkData, mergeData, recodeData, recodeMnr,
+           aggregateData, scoreData, writeSpss, collapseMissings, suppressErr,
+           rename, recodedData, addLeadingZeros, truncateSpaceChar, verbose), checkmate::assert_logical, len = 1)
+  checkmate::assert_numeric(breaks, null.ok = TRUE)
+  checkmate::assert_numeric(nMbi, len = 1, lower = 1)
+  checkmate::assert_character(recodeErr, len = 1)
+
+  lapply(c(rotation.id, newID), checkmate::assert_character, len = 1, null.ok = TRUE)
+  checkmate::assert_character(oldIDs, null.ok = TRUE)
+  checkmate::assert_list(missing.rule, names = "named")
```
Comment (sachseka) [inline R/automateDataPreparation.R:25]: interesting argument (names). I didn't know this before. here we could even use/should use names="unique"

**Review by sachseka — APPROVED** [review]
Just a very minor thing: For checking the missing.rule, we should use names="unique". after this, branch can be merged directly

**Pull request merged**